service.rb

```diff
@@ -27,6 +27,26 @@ def initialize(port)
         end
         response_message << "</ul>\n"
         response_message << daily_data_form
+      when ["POST", "/add-daily-data"]
+        content_type = "text/html"
+
+        while not client.eof?
+          line = client.readline
+          if line.include?("Content-Length: ")
+            line.slice!("Content-Length: ")
+            content_length = line.to_i
+            break
+          end
+        end
+
+        while not client.eof?
+          line = client.readline
+          if line == "\r\n"
```

Comment:
Nitpick: it would be slightly nicer if we didn’t have to mention the magic string `"\r\n"` explicitly here. What changes are necessary to make it possible to ask a more semantically-meaningful question like `line.empty?` instead?

Reply:
Didn't occur to me that `line.empty?` does the same check. 👍

Reply:
As discussed: me either! I was trying to encourage you to do `line = client.readline.chomp` here so that the whitespace got consumed, but TIL that `#empty?` ignores whitespace anyway! 😊

Reply:
Hold on, I just tried it and it’s not true:

```ruby
% irb
>> "\r\n".empty?
=> false
```

Does it behave differently on your machine? Or has one of us (presumably me) got confused somehow? I thought you told me earlier that `"\r\n".empty?` was `true` but perhaps I misunderstood.